### DISCUSSION TITLE
Card layout options - Allow select images from nested collections at image source

### DISCUSSION THREAD

**Discussion by sidartaveloso** [directus/directus#9683] [Feature Requests]
It would be great if we could select the image from nested collections.

Example:

```typescript
Client {
  id: UUID,
  person: UUID // (m2o -> person)
}

Person {
  id: UUID,
  name: string,
  avatar: UUID (Image)
}
```

![image](https://user-images.githubusercontent.com/144222/141148838-c45ea917-8a6f-499e-bb7d-be5a67dd972d.png)

**Comment by rijkvanzanten** [comment]
This should be doable by updating this:

https://github.com/directus/directus/blob/82789804fe7b1607c9db5aadab6660781adf2e7a/app/src/layouts/cards/index.ts#L42-L56

to rely on the useFieldTree composable, like in here for example:

https://github.com/directus/directus/blob/82789804fe7b1607c9db5aadab6660781adf2e7a/app/src/interfaces/_system/system-field-tree/system-field-tree.vue#L62

which can then be fed to the v-select it's already using for that option here:

https://github.com/directus/directus/blob/82789804fe7b1607c9db5aadab6660781adf2e7a/app/src/layouts/cards/options.vue#L4

The one thing we'll have to look into is that we'd want to hide / disable non-file fields, to prevent people from accidentally selecting a non-file field as the "image source" option. I believe the `useFieldTree` composable has a filter callback function that we can use for that purpose. @Oreilles could you confirm the usage eof that filter callback for this purpose?

**Reply by oreilles** [reply]
The `filter` options in `useFieldTree` definetly fit that purpose 👍

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
